src/source/spatial.rs

```diff
@@ -16,6 +15,13 @@ where
     input: ChannelVolume<I>,
 }
 
+fn dist(a: [f32; 3], b: [f32; 3]) -> f32 {
+    a.iter().zip(b.iter())
+        .map(|(a, b)| (a - b) * (a - b))
+        .sum::<f32>()
+        .sqrt()
```

Comment:
Technically you can save a lot of work and processing time by not using a `.sqrt()`, and adjusting the `set_positions` function... (see next message)